tests/functional/test_install.py

```diff
@@ -228,6 +228,32 @@ def test_install_editable_from_git(script, tmpdir):
     result.assert_installed('pip-test-package', with_files=['.git'])
 
 
+def test_install_editable_uninstalls_existing(data, script, tmpdir):
+    """
+    Test that installing an editable uninstalls a previously installed
```

Comment:
a second test that deals with a non-vcs "-e <path>" install would be ideal, to confirm that also works as we'd want.